samples/SampleStartups/StartupConfigureOptions.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+
+// Note that this sample will not run. It is only here to illustrate usage patterns.
+
+namespace SampleStartups
+{
+    public class StartupConfigureOptions : StartupBase
+    {
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public override void Configure(IApplicationBuilder app)
+        {
+            app.Run(async (context) =>
+            {
+                await context.Response.WriteAsync("Hello World!");
+            });
+        }
+
+        // Entry point for the application.
+        public static void Main(string[] args)
+        {
+            var host = new WebHostBuilder()
+                //.UseKestrel()
+                .UseFakeServer()
+                .UseStartup<StartupConfigureOptions>()
+                // Register transient IConfigureOptions
+                .ConfigureOptions<ConfigureOptions>()
```

Comment:
these would just become `.ConfigureServices(services => services.ConfigureOptions<ConfigureOptions>())`

Reply:
Not much point in adding tests in Hosting if there's no Hosting API to test. You have tests in Options right?

Reply:
Yep, no point to touch hosting if we don't add any new APIs, which is why I sent this PR before actually adding the tests to hosting :)